beacon_node/store/src/hot_cold_store.rs

```diff
@@ -2906,95 +2916,14 @@ pub fn migrate_database<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>>(
-                                    .push(StoreOp::DeleteSyncCommitteeBranch(prev_block_root));
-                            } else {
-                                hot_db_ops
-                                    .push(StoreOp::DeleteSyncCommitteeBranch(*entry_block_root));
-                            }
-                        }
-                        Entry::Vacant(v) => {
-                            v.insert((*slot, *block_root));
-                        }
-                    }
-                } else if epochs_with_boundary_blocks.contains(&slot.epoch(E::slots_per_epoch()))
-                    && !epochs_with_boundary_blocks
-                        .contains(&(slot.epoch(E::slots_per_epoch()) + Epoch::new(1)))
-                {
-                    // We have a boundary block for this epoch, but not the next epoch. We keep track of the largest non-missed slot
-                    // in this epoch to use as the checkpoint slot for the next epoch while staging all other sync committee branches for deletion.
-                    let epoch = slot.epoch(E::slots_per_epoch());
-                    match checkpoint_slots_on_non_epoch_boundaries.entry(epoch) {
-                        Entry::Occupied(mut occupied_entry) => {
-                            let (entry_slot, entry_block_root) = occupied_entry.get();
-                            if slot > entry_slot {
-                                let (_, prev_block_root) =
-                                    occupied_entry.insert((*slot, *block_root));
-                                hot_db_ops
-                                    .push(StoreOp::DeleteSyncCommitteeBranch(prev_block_root));
-                            } else {
-                                hot_db_ops
-                                    .push(StoreOp::DeleteSyncCommitteeBranch(*entry_block_root));
-                            }
-                        }
-                        Entry::Vacant(v) => {
-                            v.insert((*slot, *block_root));
-                        }
-                    }
-                } else {
-                    // There was a missed block at this epochs boundary slot, but not at the next epoch boundaries slot. Since
-                    // the boundary slot for this epoch will end up being at the previous epoch, we can safely stage all
-                    // sync committee branches for deletion.
-                    hot_db_ops.push(StoreOp::DeleteSyncCommitteeBranch(*block_root));
-                }
-            });
-    } else {
-        non_epoch_boundary_blocks
-            .iter()
-            .for_each(|(_, block_root)| {
-                hot_db_ops.push(StoreOp::DeleteSyncCommitteeBranch(*block_root));
-            });
-    }
+    // Prune sync committee branch data for all non checkpoint block roots.
+    // Note that `non_checkpoint_block_roots` should only contain non checkpoint block roots
+    // as long as `finalized_state.slot()` is at an epoch boundary.
```

Comment:
What happens otherwise?

Reply:
take the following example

current split slot: Epoch A slot 0
finalized_state.slot(): Epoch C slot 31

as we go through the pruning logic we'll end up pruning Epoch C slot 31. If Epoch D slot 0 happens to be a skipped slot we will have deleted the sync committee branch for a checkpoint slot.

This should never happen because we make this check before pruning

``` rust
if finalized_state.slot() % E::slots_per_epoch() != 0 {
   return Err(HotColdDBError::FreezeSlotUnaligned(finalized_state.slot()).into());
}
```

So as long as this invariant doesn't change, we should be ok

Reply:
Makes sense, please document in the code comment

Reply:
a202eac